pkg/api/v1alpha1/tinkerbellmachineconfig.go

```diff
@@ -125,3 +127,27 @@ func validateTinkerbellMachineConfig(config *TinkerbellMachineConfig) error {
 
 	return nil
 }
+
+func setTinkerbellMachineConfigDefaults(machineConfig *TinkerbellMachineConfig) {
+	if len(machineConfig.Spec.Users) <= 0 {
+		machineConfig.Spec.Users = []UserConfiguration{{}}
+	}
+
+	if len(machineConfig.Spec.Users[0].SshAuthorizedKeys) <= 0 {
+		machineConfig.Spec.Users[0].SshAuthorizedKeys = []string{""}
+	}
+
+	if machineConfig.Spec.OSFamily == "" {
+		logger.Info("Warning: OS family not specified in machine config specification. Defaulting to Bottlerocket.")
+		machineConfig.Spec.OSFamily = Bottlerocket
+	}
+
+	if len(machineConfig.Spec.Users) == 0 || machineConfig.Spec.Users[0].Name == "" {
```

Comment:
We don’t need to check len of users here again right? 
Since if it was 0 we set a default above.

Reply:
We do because if the users array is an empty array, the 2nd condition machineConfig.Spec.Users[0].Name == "" could throw a panic error as it does not have a `Name` field in the empty array.